apiserver/facades/agent/caasprovisioner/provisioner.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasprovisioner
+
+import (
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/state/watcher"
+)
+
+type API struct {
+	auth      facade.Authorizer
+	resources facade.Resources
+
+	state CAASProvisionerState
+}
+
+// NewStateCAASProvisionerAPI provides the signature required for facade registration.
+func NewStateCAASProvisionerAPI(ctx facade.Context) (*API, error) {
+
+	authorizer := ctx.Auth()
+	resources := ctx.Resources()
+	return NewCAASProvisionerAPI(resources, authorizer, ctx.State())
+}
+
+// NewCAASProvisionerAPI returns a new CAASProvisionerAPI facade.
+func NewCAASProvisionerAPI(
+	resources facade.Resources,
+	authorizer facade.Authorizer,
+	st CAASProvisionerState,
+) (*API, error) {
+	if !authorizer.AuthApplicationAgent() && !authorizer.AuthController() {
```

Comment:
why is this facade serving both application agents and controllers? makes sense for it to serve controllers only I think?

Reply:
yeah, i left it in by mistake